popup.js

```diff
@@ -12,24 +17,63 @@ fileSelector.onchange = function () {
 
     reader.onload = function (e) {
         console.log(e.target.result)
-        let json = JSON.parse(e.target.result)
-        console.log(json["primaryColor"])
-        let schemeCode, scheme, color
-        for (let i = 0; i < colorScheme.length; i++) {
-            scheme = colorScheme[i]
-            color = json[colorScheme[i]]
-            schemeCode = styleTernary + scheme + ': ' + color + ' !important;")'
-            tabScript(schemeCode);
-
-            chrome.storage.sync.set({
-                [scheme]: color
-            });
-        }
-        location.reload()
+        importScheme(e.target.result)
     };
     reader.readAsText(file)
 }
 
+const IMPORT_EXPORT_MODE = Object.freeze({
+    IMPORT: 'import',
+    EXPORT: 'export'
+})
+
+let importExportMode = null
+
+/**
+ * Set basic import/export container visibility
+ * @param {boolean} isVisible 
+ * @returns 
+ */
+const setBasicImportExportVisibility = (isVisible) => basicImportExportContainer.style.display = isVisible ? 'block' : 'none'
+/**
+ * Toggle visiblity of the basic import/export container
+ */
+const toggleBasicImportExport = () => {
+    const isHidden = basicImportExportContainer.style.display === 'none'
+    setBasicImportExportVisibility(!isHidden)
+}
+
+/**
+ * Set the import export color mode
+ * @param {'import' | 'export'} mode 
+ */
+const setImportExportMode = (mode) => {
+    const label = mode === IMPORT_EXPORT_MODE.IMPORT ? 'Save' : "Close"
+    importExportMode = mode;
+    importExportActionButton.textContent = label
+    if(mode === IMPORT_EXPORT_MODE.EXPORT){
+        basicImportExportInput.focus()
+    }else if(mode === IMPORT_EXPORT_MODE.IMPORT){
+        basicImportExportInput.value = ''
+    }
+}
+
+const importScheme = (data) => {
+    let json = JSON.parse(data)
+    console.log(json["primaryColor"])
+    let schemeCode, scheme, color
+    for (let i = 0; i < colorScheme.length; i++) {
+        scheme = colorScheme[i]
+        color = json[colorScheme[i]]
```

Comment:
hey any chance we can update this line to this ternary:
`color = json[scheme] ? json[scheme] : colorDefaults[i]`

My JSON generating pattern does more work than it needs to and hid a potential bug here. I have a better fix in mind but this will keep things working for now